src/main/java/org/junit/runner/OrderWith.java

```diff
@@ -0,0 +1,26 @@
+package org.junit.runner;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.runner.manipulation.Ordering;
+
+/**
+ * When a test class is annotated with <code>&#064;OrderWith</code> or extends a class annotated
+ * with <code>&#064;OrderWith</code>, JUnit will order the tests in the test class (and child
+ * test classes, if any) using the ordering defined by the {@link Ordering} class.
+ *
+ * @since 4.13
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface OrderWith {
```

Comment:
Sorry for the long delay, but I hope that we can release JUnit 4.13 in 2017 and I'd like to include this pull.

I updated the code to pass a context object that allows you to access the top-level description
of the class annotated with `@OrderWith`. This will handle David's use-case of " run the tests from the fastest to the slowest, based on a record of previous runtimes kept in a file". The filename could
either be specified as an annotation on the class annotated with `@OrderWith` or it could be
derived from the class name of the class annotated with `@OrderWith`